qds_sdk/commands.py

```diff
@@ -254,6 +254,98 @@ def parse(cls, args):
         v["command_type"] = "HiveCommand"
         return v
 
+class SparkCommand(Command):
+	
+	#{"program":prog,"language":"python","command_type":"SparkCommand","label":"praveen-spark"} 
+    usage = ("sparkcmd <submit|run> [options]")
+    allowedlandlist = ["python", "scala"]
+    optparser = GentleOptionParser(usage=usage)
+    optparser.add_option("--program", dest="program", help="program code")
+    
+    optparser.add_option("--cmdline", dest="commandline", help="command line for Spark")
+        
+    optparser.add_option("-f", "--script_location", dest="script_location",
+                         help="Path where spark program to run is stored. Can be S3 URI or local file path")
+
+    optparser.add_option("--macros", dest="macros",
+                         help="expressions to expand macros used in query")
+	
+    optparser.add_option("--tags", dest="tags",
+                         help="comma-separated list of tags to be associated with the query ( e.g., tag1 tag1,tag2 )")
+
+    optparser.add_option("--sample_size", dest="sample_size",
+                         help="size of sample in bytes on which to run query")
+
+    optparser.add_option("--cluster-label", dest="label", help="the label of the cluster to run the command on")
+	
+    optparser.add_option("--language", dest="language", help="the language used in the spark program, supported languages are scala and python")
+    
+    optparser.add_option("--label", dest="label_program",help="the label for the program")
+	
+    optparser.add_option("--notify", action="store_true", dest="can_notify", default=False, help="sends an email on command completion")
+
+    optparser.add_option("--name", dest="name", help="Assign a name to this query")
+           
+
+    @classmethod
+    def parse(cls, args):
+        """
+        Parse command line arguments to construct a dictionary of command
+        parameters that can be used to create a command
+
+        Args:
+            `args`: sequence of arguments
+
+        Returns:
+            Dictionary that can be used in create method
+
+        Raises:
+            ParseError: when the arguments are not correct
+        """
+        allowedlanglist = ["python", "scala"]
+        try:
+            (options, args) = cls.optparser.parse_args(args)
+            if options.program is None and options.script_location is None and options.commandline is None:
+                raise ParseError("One of script location or program or commandline must be specified", cls.optparser.format_help())
+        except OptionParsingError as e:
+            raise ParseError(e.msg, cls.optparser.format_help())
+        except OptionParsingExit as e:
+            return None
+
+        if options.script_location is not None:
```

Comment:
May be simpler to do something like below. I think such fine grained errors are probably not needed.
if (script_location xor program xor cmdline) {
  error(only one of script_location, program, cmdline can be specified.
}